library/model/book.py

```diff
@@ -168,3 +136,68 @@ def serialize(self, book: Book, format: str):
             return et.tostring(book_info, encoding="Unicode")
         else:
             raise ValueError(format)
+
+
+class ElectronicBook(Book):
+    pages: int
+
+    def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn):
+        super(ElectronicBook, self).__init__(title, authors, publisher, pub_date, genres, isbn)
+        self.pages = pages
```

Comment:
You can define Book and all its subclasses as dataclasses and these __init__'s willbe automatically implemented for you

Reply:
fascinating